felix/bpf/bpf_syscall.go

```diff
@@ -269,8 +269,16 @@ func checkMapIfDebug(mapFD MapFD, keySize, valueSize int) error {
 		if keySize != mapInfo.KeySize {
 			log.WithField("mapInfo", mapInfo).WithField("keyLen", keySize).Panic("Incorrect key length")
 		}
-		if valueSize >= 0 && valueSize != mapInfo.ValueSize {
-			log.WithField("mapInfo", mapInfo).WithField("valueLen", valueSize).Panic("Incorrect value length")
+		// The actual size of per cpu maps is equal to the value size * number of cpu
+		if mapInfo.Type == unix.BPF_MAP_TYPE_PERCPU_HASH || mapInfo.Type == unix.BPF_MAP_TYPE_PERCPU_ARRAY ||
+			mapInfo.Type == unix.BPF_MAP_TYPE_LRU_PERCPU_HASH || mapInfo.Type == unix.BPF_MAP_TYPE_PERCPU_CGROUP_STORAGE {
+			if valueSize >= 0 && valueSize != mapInfo.ValueSize*runtime.NumCPU() {
```

Comment:
Is `NumCPU()` the right number?  ISTR that the map gets sized according to some kernel-internal view of hte max number of CPUs that your motherboard _could_ support.

Reply:
hmmm... so if a system has two cpu sockets but only one is used, then the per cpu map should be provisioned for both? It not easy to find such a system for test.

Reply:
As @fasaxc suggested We should either use the libbpf function or get the value from /sys/devices/system/cpu/possible ourselves

Reply:
IIUC, felix does not have access to read `/sys/devices/system/cpu/possible` (is that correct?). If that's correct, does it mean the only option is by using libbpf?

Reply:
That library also read that file.

Reply:
Why do you think we don't have access to that file?

Reply:
Right, I thought of something else. Will implement by reading `/sys/devices/system/cpu/possible`

Reply:
Sure you don't want to just expose the libbpf function?  That should be easier given all the libbpf wrappers we already have.